### ISSUE TITLE
Improve cache controls for foundry items

### ISSUE THREAD

**Issue by KageJittai** [foundryvtt/foundryvtt#9642]
### User Experience

As of 11.302 all files use `Cache-Control: no-cache` headers to prevent browser caching. This results in a 304 round-trip whenever the file is attempted to be loaded to check for changes. My understanding is that his was changed to prevent user issues when modules or systems were updated.

While this is very likely a good catch-all blanket policy, especially for code file, due to the large number of files, notably icon images, any foundry page loads results in dozens to hundreds of round-trip requests to check the version of files. I think a smarter cache policy can be used.

This request comes with three recommendations for improvements:

1. Common core Foundry images and audio files, such as those provided by the core foundry system such as in the `ui/` or `icons/` should be allowed to have longer cache durations. I believe this is the minimal that should be done.
2. Consider a file type based cache policy. While JS/CSS files are likely the most commonly to cause an issue when cache is used. Images and sounds are less likely to change or cause usability issues if they are changed, and on average are of a much larger size. 
3. If a file based policy is not used, the ability for a module to control their own caching would be useful. Some modules include popular modules like Dice So Nice use a large number of texture files. This could be configured inside of the module.json file or by using an apache style `.htaccess` file. Which might already be supported in the Express/Node HTTP Server system, or would need to be implemented by foundry.

**Comment by Fyorl** [comment]
Reasonably certain we didn't make a change to this in v11, and we've used `no-cache` since v9 or something? Maybe the move to `PIXI.Assets` instead of our own loader has meant `no-cache` is being used where we inadvertently skipped it before?

**Comment by rexmortus** [comment]
Yes, please. I think most of the UI images could have some cache life to prevent their re-render in certain mobile browsers (looking at you Safari!)
